### PR TITLE
Updated build tools and readme for android.

### PR DESCRIPTION
**Pull request by akallabeth**
* Updated gradle and android build tools
* Updated readme
* Replaced ```int``` with ```long``` for 64bit architecture support.
* Fixed compiler warnings

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/core/connection.c [hunk 1]
```diff
@@ -773,7 +773,7 @@ BOOL rdp_client_connect_auto_detect(rdpRdp* rdp, wStream* s)
 		{
 			if (channelId == rdp->mcs->messageChannelId)
 			{
-				UINT16 securityFlags;
+				UINT16 securityFlags = 0;
```
Comment (bmiklautz) [inline libfreerdp/core/connection.c]: Was this a compiler warning? - As the securityFlags are always initialized because they are read below.

Comment (akallabeth) [inline libfreerdp/core/connection.c]: @bmiklautz Yes, there was a warning during android compile. I've just initialized it for every case where the read / write functions are used.

**Comment by akallabeth** [comment]
@bmiklautz ok, done

**Pull request merged**